[NEO-78835]: fix extraHttpHeaders issue

pankajkrk8 opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description

The extraHttpHeaders passed to the connection parameters which is used to initialise the client, is not present in the XtkJobInterface's function and FileUploader APIs

```
const connectionParameters = sdk.ConnectionParameters.ofBearerToken(targetDomain, imsToken, {extraHttpHeaders: {testHeader: 'headerValue'}});
const clientSdk = await sdk.init(connectionParameters).catch(() => false);
await clientSdk.logon();

const job = await clientSdk.jobInterface({
  xtkschema: "nms:dataModel",
  method: "write",
  jobId: name,
  args: dataModelJSON,
});

// This API call does not have the extraHttpHeaders in the headers.
await job.submitSoapCall();
```

Also, the submit and execute APIs does not have the extraHttpHeaders in the header.
Although, all the other methods like getStatus, getResult, cancel, pause, waitJobCancelled and hasWarning already support the extraHttpHeaders as they are being handled by the Proxy clientHandler.

Similarly, we have this issue in the fileUploader APIs.

## Related Issue

<!--- This project only accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here: -->

## Motivation and Context

<!--- Why is this change required? What problem does it solve? -->

## How Has This Been Tested?

<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## Screenshots (if appropriate):

## Types of changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

mkiki reviewed (changes requested):
I see in the PR that you have modified a couple of tests to pass connectionParameters: { _options: {}} }. Could you explain why? Normally unit tests serve as regression tests, so any changes in the tests is a sign that there could be a regression.

Also, I see don't see any new tests regarding this PR, could you add some? We'd like to see tests which asserts that the extra headers are correctly passed now, i.e. those tests should fail with the previous code and succeed with the new code

pankajkrk8 commented:
> I see in the PR that you have modified a couple of tests to pass connectionParameters: { _options: {}} }. Could you explain why? Normally unit tests serve as regression tests, so any changes in the tests is a sign that there could be a regression.
> 
> Also, I see don't see any new tests regarding this PR, could you add some? We'd like to see tests which asserts that the extra headers are correctly passed now, i.e. those tests should fail with the previous code and succeed with the new code

The connectionParameter is something which will always be present in the client object. But the client mocking in XtkJob test cases did not have it, so I added it. If we use the [MockClient](https://github.com/adobe/acc-js-sdk/blob/master/test/mock.js#L40), then we do not have to add this required parameter.

Also, I have added the test cases for extra headers.

mkiki reviewed (approved):
Thanks for the quick changes
